### PR TITLE
Migrate to K2, add Qodana

### PR DESCRIPTION
**Pull request by Daoortor**
Detekt is not yet supported in kotlin 2.0, so Qodana was added instead.

### CODE REVIEW

File: .github/workflows/gradle-build-with-detekt.yml [hunk 1]
```diff
@@ -1,6 +1,7 @@
 name: Gradle Build With Detekt
 
-on: [push, pull_request]
+# Detekt doesn't work with Kotlin 2.0 for now, so it's disabled by default
+on: [workflow_dispatch]
```
Comment (mikrise2) [inline .github/workflows/gradle-build-with-detekt.yml:4]: Here you have disabled the action execution on pull request or push, I'm not sure if this should be done. Especially I don't understand why **Detekt** won't run. In order to change the behavior of the action during execution, you need to change `jobs`.

File: .github/workflows/qodana.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      checks: write
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}  # to check out the actual pull request commit, not the merge commit
+          fetch-depth: 0  # a full history is required for pull request analysis
+      - name: 'Qodana Scan'
+        uses: JetBrains/qodana-action@v2024.1
+        env:
+          QODANA_TOKEN: ${{ secrets.QODANA_TOKEN }}
```
Comment (mikrise2) [inline .github/workflows/qodana.yml:26]: It looks very much like the file is almost entirely generated. It seems that we don't have branches that are specified in the config.... I suggest rewriting this file

**Review by mikrise2 — CHANGES_REQUESTED** [review]
Some comments

**Review by nbirillo — CHANGES_REQUESTED** [review]
So as I can see in the merged PR you have not updated the compiled sources in the server resources, please do it. Also the workflow has not been run...

**Pull request merged**